File: output/gpt_4o/avatar/direct_translation/python/java/codeforces_624_B.java

import java.util.*;

public class codeforces_624_B {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = Integer.parseInt(sc.nextLine().trim());
        int[] a = Arrays.stream(sc.nextLine().trim().split(" ")).mapToInt(Integer::parseInt).toArray();
        int pre = Integer.MAX_VALUE;
        int ans = 0;
        Arrays.sort(a);
        for (int j = n - 1; j >= 0; j--) {
            ans += Math.max(0, Math.min(pre - 1, a[j]));
            pre = Math.max(0, Math.min(pre - 1, a[j]));
        }
        System.out.println(ans);
    }
}

File: output/qwen_2.5_72b_instruct/avatar/direct_translation/java/python/successful_translation/codeforces_624_B.py

import sys

n = int(input())
ans = list(map(int, input().split()))
ans.sort()
count = ans[-1]
for i in range(n - 2, -1, -1):
    if ans[i] < ans[i + 1]:
        count += ans[i]
    else:
        while ans[i] >= ans[i + 1]:
            ans[i] -= 1
        if ans[i] > 0:
            count += ans[i]
print(count)

